### PR TITLE
lantiq: new subtarget and two device updates

### PR DESCRIPTION
**Pull request by mcmilk**
This patchset will re-add the Amazon-SE SoC to LEDE.

This SoC is used on a lot low budget ADSL2+ Modem devices.
The two devices Netgear DGN1000B and Allnet ALL0333CJ are the first ones in this target.

### CODE REVIEW

**Comment by blogic** [comment]
@hauke @mkresin @mcmilk 

do we need to make changes to the dsl packages ? apart form that LGTM

**Comment by mcmilk** [comment]
@blogic 

the dsl packages compile fine currently. But the modem firmware is not correctly loaded by the dsl_cpe_control binary. I am trying to find the reason for that with some debugging today...

here is some output, if someone has also interest in it:
[   29.405962] [IFX_MEI_RunAdslModem 1369]: Modem failed to be ready!
[   29.410889] [IFX_MEI_Ioctls 2564]: IFX_MEI_RunAdslModem() error...
[   32.372792] [IFX_MEI_RunAdslModem 1369]: Modem failed to be ready!
[   32.377832] [IFX_MEI_Ioctls 2564]: IFX_MEI_RunAdslModem() error...
[   32.704550] random: nonblocking pool is initialized
[   34.676645] [IFX_MEI_RunAdslModem 1369]: Modem failed to be ready!
[   34.681547] [IFX_MEI_Ioctls 2564]: IFX_MEI_RunAdslModem() error...

PS: I had found one double define in this file:
`package/kernel/lantiq/ltq-atm/src/ifxmips_atm_amazon_se.c`

- EMA_CMD_BASE_ADDR and some other definitions are defined two times...
- when I know the correct one, I will submit a fix for it

File: target/linux/lantiq/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -183,4 +190,3 @@ ucidef_set_interface_wan "$interface_wan" "pppoe"
 board_config_flush
 
 exit 0
-
```
Comment (mkresin) [inline target/linux/lantiq/base-files/etc/board.d/02_network]: Please keep this newline

File: target/linux/lantiq/dts/ALL0333CJ.dts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			led-internet = &online_green;
+		};
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x1000000>; // 16MB
+	};
+
+	// does not work at fpi@10000000 /TR
```
Comment (mkresin) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: drop this and all following comments

File: target/linux/lantiq/dts/ALL0333CJ.dts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	 *  3=ethernet      (1=off, 0=on)
+	 *  4=online red    (1=off, 0=on) - right most
+	 * 12=online green  (1=off, 0=on) - right most
+	 * 13=power         (1=red, 0=green) - left most
+	 */
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		/* POWER: two led's: 1x green, 1x red */
```
Comment (mkresin) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: please add red:power to the dts

File: target/linux/lantiq/dts/ALL0333CJ.dts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		power: power {
+			label = "all0333cj:green:power";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+
+		lan: lan {
+			label = "all0333cj:green:lan";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
```
Comment (mkresin) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: Why does this led has default-state keep?

**Comment by mkresin** [comment]
Please document the rootfs validation workaround of the Allnet in the commit message. As far as I remember some u-boot env variables need to be changed to get it working => document the changes in the commit message.

TO find the reset button pin, have a look at the [gpio scripts mentioned in the OpenWrt wiki](https://wiki.openwrt.org/doc/hardware/port.gpio#finding_gpio_pins_on_the_pcb).

File: target/linux/lantiq/dts/ALL0333CJ.dts (hunk 3 above)
Comment (mkresin) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: ignore this one. failed to read your comment. The led is either green or red. But never off.

File: target/linux/lantiq/base-files/etc/board.d/02_network (hunk 1 above)
Comment (mcmilk) [inline target/linux/lantiq/base-files/etc/board.d/02_network]: There are two enters in the end. Is this a wanted feature ?

File: target/linux/lantiq/dts/ALL0333CJ.dts (hunk 2 above)
Comment (mcmilk) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: Yes, will do this.

File: target/linux/lantiq/dts/ALL0333CJ.dts (hunk 4 above)
Comment (mcmilk) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: This was a test, must be removed. Nice catch, thank you.

File: target/linux/lantiq/dts/ALL0333CJ.dts (hunk 3 above)
Comment (mcmilk) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: Yes, this is sth. like an double LED, so I did this extra comment there.

**Comment by mcmilk** [comment]
Whats the best way to edit the commit message of the "lantiq: add Allnet ALL0333CJ support" patch 54f5cff ?

I see 3 possibilities, which should all work more or less good:
1. `git pick 54f5cff + git rebase` (don't know how to do currently)
2. `git push -f` (will delete comments in PR I think)
3. `git revert` + `git commit` with new commit message (will delete comments in PR I think)

File: target/linux/lantiq/base-files/etc/board.d/02_network (hunk 1 above)
Comment (mkresin) [inline target/linux/lantiq/base-files/etc/board.d/02_network]: no it isn't a wanted feature. but that change is unrelated to your commit and should be fixed a separate (cleanup) commit.

File: target/linux/lantiq/dts/ALL0333CJ.dts (hunk 3 above)
Comment (mkresin) [inline target/linux/lantiq/dts/ALL0333CJ.dts]: I would add it to the commit message. but no strong opinion on this. for sure it should be documented.

**Comment by bobafetthotmail** [comment]
I used **git reset HEAD~X** where X is the number of commits to delete. This will not delete files, only git commits. I delete all commits I made after the commit I need to change.
Then I use **git commit --amend** to change the commit message of the commit I wanted to change.
Then **git add -i** to add other modified files to git tracked list with its interface and **git commit** to make the commit(s) again.
Then **git push -f** to send the changes over to github and in the PR.

While you are at it, you might want to squash some of your commits in a single one.

Afaik comments here on GIthub are not deleted, they are only tagged as "obsoleted" after the commit they were referring to disappears.
Messages in the PR remain visible.

**Comment by mkresin** [comment]
> Whats the best way to edit the commit message of the "lantiq: add Allnet ALL0333CJ support" patch 54f5cff ?

If it is the last commit then ```git commit --amend```.

Otherwise

```git rebase -i HEAD~5```

change ```pick``` to ```reword``` to edit the commit message. change ```pick``` to ```edit``` to edit the whole commit.

Please squash the new added commits in to the existing ones. You can reorder the commits in that rebase list. Change ```pick``` to ```squash``` to merge the commit with the one one line above.

Do a ```git push --force``` to overwrite your github branch.

**Comment by mcmilk** [comment]
@mkresin
`git rebase -i HEAD~5` is magic. **Thank you**, for that hint.

**Comment by mcmilk** [comment]
I have put together the "lantiq: add Allnet ALL0333CJ support" patch and its fixes.

**Comment by mkresin** [comment]
@mcmilk Hey Tino, can you give me an update? What does work, what does still not work? Should we merge this PR anyway or would you like to fix the outstanding issues first?

**Comment by mcmilk** [comment]
@mkresin Hello Mathias, sorry for the delay, I had overwritten my uboot on the test modem ;)
But I got a Hot Air Gun Rework and an TL866CS Programmer now ... so the work can restart again...

I also bought these adsl modems for 4€ on ebay: 
1) EasyBox 904 xDSL
2) EasyBox 803 Vodafone

The 803 Box is Danube based and should work out of the box, but it does not sync... I will write more details about this 803 box on the mailing list...

The patch will be reworked (today or tomorrow) and should be okay then.
The modem syncing is maybe another problem. Thats my current state...

**Comment by mcmilk** [comment]
@mkresin, I have forwarded my own branch to the current master and another commit for fixing gptu things. I didn't resolve the dsl modem issue. AFAIK, danube and amazon-se seem to be broken... but not by my patches, also beforce :-(

From my side, this patchset is ready for integration into current LEDE.

Open issues: Fix dsl sync, but this will required more time... which I currently does not have for it.

**Comment by mcmilk** [comment]
Ping

**Comment by hauke** [comment]
This looks good to me, except for the one change.
@blogic  and @mkresin  what do you think?

**Comment by mkresin** [comment]
LGTM

**Comment by hauke** [comment]
Thank you for the pull request, I merged it with some small changes.

File: package/kernel/lantiq/ltq-adsl-mei/src/lantiq_mei.c [hunk 5]
```diff
@@ -79,13 +79,33 @@
...
+#ifdef CONFIG_AMAZON_SE
+# define LTQ_MEI_INT             (INT_NUM_IM2_IRL0 + 9)
+# define LTQ_MEI_DYING_GASP_INT  (INT_NUM_IM2_IRL0 + 11)
+# define LTQ_USB_OC_INT          (INT_NUM_IM2_IRL0 + 20)
+#endif
+
+#ifdef CONFIG_AR9
+# define LTQ_MEI_INT             (INT_NUM_IM1_IRL0 + 23)
+# define LTQ_MEI_DYING_GASP_INT  (INT_NUM_IM1_IRL0 + 21)
+# define LTQ_USB_OC_INT          (INT_NUM_IM4_IRL1 + 28)
```
Comment (mcmilk) [inline package/kernel/lantiq/ltq-adsl-mei/src/lantiq_mei.c]: The `#elif defined(CONFIG_SOC_AR9)` block on line 1335 was this:
`#define IFXMIPS_USB_OC_INT0 (INT_NUM_IM4_IRL1 + 28)`

... I just moved it upwards to get rid of the lot `#ifdef's` there ... it made sense, to move them onto top ;-)
